arm9/source/game/codelzss.c

```diff
@@ -44,6 +46,10 @@ u32 DecompressCodeLzss(u8* code, u32* code_size, u32 max_size) {
     
     // main decompression loop
     while ((ptr_in > comp_start) && (ptr_out > comp_start)) {
+        if ((counter++ == 0) &&
+            !ShowProgress(data_end - ptr_out, data_end - data_start, "Decompressing .code...") &&
```

Comment:
Given that this typically takes almost no time at all (you may convince me otherwise), does it really make sense? This is the only place the progress bar is used and also the only reason for the `ui.h` dependency (which makes the code less portable). Also keep in mind that the progress bar has to be initialized / reinitialized with `current == 0`.

EDIT: yeah, I see, the prgress bar is used in that other function, too. In general, only introduce dependencies like `ui.h` it it is really required. This one is most likely justified for the compression function.